Automattic/woocommerce-services
# Add Stripe banner for connecting existing accounts

In the event that deferred Stripe account creation fails due to an account already existing with that email address, this banner will now show up in wp-admin:

<img width="858" alt="screen shot 2018-08-30 at 3 26 28 pm" src="https://user-images.githubusercontent.com/1867547/44874422-1cd40280-ac69-11e8-81d4-ecd2ef10c574.png">

Clicking "Connect" will initiate the OAuth flow on stripe.com, starting with requesting certain business information if it's missing, and culminating in:

<img width="782" alt="screen shot 2018-08-30 at 6 31 25 pm" src="https://user-images.githubusercontent.com/1867547/44882823-ff139700-ac82-11e8-9023-ccdc4bb4e6bd.png">

(Note that if you click api.woocommerce.com, you see: https://cloudup.com/cFf9cnOhm5q)

Upon completing this flow, the user arrives at their Stripe settings page with the API keys filled in and the banner gone ([screenshot](https://cloudup.com/cvPc_EQGAMk)).

Fixes broken flow for the existing account case, which is now explicitly supported as of https://github.com/woocommerce/woocommerce/pull/21224. Should address https://github.com/Automattic/woocommerce-services/issues/1319 in part. Based on static design in p7bbVw-2dB-p2.

To test:
- In the Payments step of the onboarding wizard (`/wp-admin/index.php?page=wc-setup&step=payment`), enable Stripe, check the checkbox, enter the email for an existing Stripe account that you can log into, and "Continue"
- Back in the wp-admin dashboard, see the Stripe banner on any of these screens: Dashboard, Plugins, WooCommerce » Settings » Payments, WooCommerce » Extensions » Payments
 click "Connect"
- After returning from Stripe, verify that the publishable key and secret key are filled ("test" keys if using staging server, "live" keys if using production server)

Concerns:
- Do any failure cases call for special handling?
- Could / should we have a special `stripe_state` value instead of adding a new `banner_stripe` option?
